src/NHibernate/Dialect/Dialect.cs

```diff
@@ -2246,6 +2248,11 @@ protected void RegisterKeyword(string word)
 			Keywords.Add(word);
 		}
 
+		protected void RegisterKeywords(params string[] keywords)
+		{
+			Keywords.UnionWith(keywords);
```

Comment:
It should be  higher performance to not generate new strings, so I made the `HashSet`'s case insensitive.